copier/cli.py

```diff
@@ -358,6 +367,59 @@ def main(self, destination_path: cli.ExistingDirectory = ".") -> int:
         return 0
 
 
+@CopierApp.subcommand("check")
+class CopierCheckSubApp(cli.Application):
+    """The `copier check` subcommand.
+
+    Use this subcommand to check if an existing subproject is using the
+    latest version of a template.
+
+    Attributes:
+        exit-code: Set [exit-code][] option.
+    """
+
+    DESCRIPTION = "Check if a copy is using the latest version of its original template"
+    DESCRIPTION_MORE = dedent(
+        """\
+        The copy must have a valid answers file which contains info
+        from the last Copier execution, including the source template
+        (it must be a key called `_src_path`).
+
+        If that file contains also `_commit` and `destination_path` is a git
+        repository, this command will do its best to determine whether a newer
+        version is available, applying PEP 440 to the template's history.
+        """
+    )
+
+    exit_code: cli.SwitchAttr = cli.SwitchAttr(
+        ["-e", "--exit-code"],
+        int,
+        default=0,
+        help=(
+            "Exit code for the command if a newer version is available, "
+            "for use in scripts."
+        ),
+    )
+
+    @handle_exceptions
+    def main(self, destination_path: cli.ExistingDirectory = ".") -> int:
+        """Call [run_check][copier.main.Worker.run_check].
+
+        Parameters:
+            destination_path:
+                Only the destination path is needed to check, because the
+                `src_path` comes from [the answers file][the-copier-answersyml-file].
+
+                The subproject must exist. If not specified, the currently
+                working directory is used.
+        """
+        try:
+            self.parent._worker(dst_path=destination_path).run_check()
+        except SubprojectOutdatedError:
+            return self.exit_code
```

Comment:
Continuing from the comment above, It'd be even better if you add a constant at the beginning of this file:

```py
EXIT_CODE_OUTDATED = 0b10
```

That's the same as `2` in base 10, but for exit codes it's more comfortable to use binary base.

Then hardcode it here. We can use this one:

```suggestion
            return EXIT_CODE_OUTDATED
```

Finally, add it to the docstring, so it's clear for the user which exit code means the project is outdated.